---
title: "Construct phyloseq object and get basic dataset stats"
author: "Natasha Dudek"
date: "21/11/2018"
output: html_document
---

This script will generate and save a phyloseq object using the following files:
1. ASV table
2. Phylogenetic tree for corresponding ASVs
3. Mapping file

We will assume that these three files are in a single working directory, specified by "path" below, and that we intend for this to be the working directory.

The script will then produce some basic stats about the dataset.

Finally, we will generate and save some ASV relative abundance csv files that will be required for plotting Figure 2A, B later on.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Set paths

```{r}
path <- file.path("/Users", "path", "to", "dir", fsep="/") # path to working dir
asv_table <- file.path(path, "filtered_asvs.csv", fsep="/") # filtered ASVs
phylogenetic_tree <- file.path(path, "tree.nwk", fsep="/") 
mapping_file <- file.path(path, "mapping_file.csv", fsep="/")
phyloseq_object <- file.path(path, "physeq_lane1.rds", fsep="/") # save physeq object as
asv_oral <- file.path(path, "asv_table_genus_oral.csv", fsep="/") # save gingival ASV table (wild sea otters)
taxa_oral <- file.path(path, "taxa_table_genus_oral.csv", fsep="/") # save gingival ASV table (wild sea otters)
asv_rectal <- file.path(path, "asv_table_genus_rectal.csv", fsep="/") # save rectal ASV table (wild sea otters)
taxa_rectal <- file.path(path, "taxa_table_genus_rectal.csv", fsep="/") # save rectal ASV table (wild sea otters)
```

Set working directory

```{r}
setwd(path)
getwd()
```

Load libraries

```{r}
library("phyloseq"); packageVersion("phyloseq")
library(ape); packageVersion("ape")
```
  
Load ASV table  

```{r load-asv-table}
so_raw <-read.csv(asv_table, header=TRUE, stringsAsFactors=FALSE)
rownames(so_raw) <- so_raw$X
so_raw$X <- NULL
otu_tab_raw <- so_raw[,1:491] # There are 491 samples in the table
OTU <- otu_table(t(otu_tab_raw), taxa_are_rows = FALSE)
```

Load taxonomic info and phylogenetic tree 

```{r}
tax_tab_raw <- as.matrix(so_raw[,492:498])
TAX <- tax_table(tax_tab_raw)
physeq <- phyloseq(OTU, TAX)
myTree <- read.tree(phylogenetic_tree)
```

Load and format mapping file with metadata

```{r}
so_metadata <-read.csv(mapping_file, header=TRUE, stringsAsFactors=FALSE)
rownames(so_metadata) <- so_metadata$SampleID
so_metadata$weight <- (as.numeric(so_metadata$weight))
so_metadata$year_captured <- (as.numeric(so_metadata$year_captured))
so_metadata$month_captured <- (as.numeric(so_metadata$month_captured))
sample_data = sample_data(so_metadata)
```

Merge OTUs, taxonomy, metadata, and phylogenetic tree into single phyloseq object  

```{r}
physeq = merge_phyloseq(physeq, sample_data, myTree)
```

Retain only samples with greater than 10*(max num reads in an extraction control) 

```{r}
xtc_only <- subset_samples(physeq, sample_type=="xtc")
thresh = max(sample_sums(xtc_only))
physeq = prune_samples(sample_sums(physeq)>(thresh*10), physeq)
physeq
```

Remove ASVs with zero reads

```{r}
physeq = filter_taxa(physeq, function(x) sum(x) > 0, TRUE)
```

Double check that there are no zero-read samples

```{r}
summary(sample_sums(physeq))
```

Double check that there are no zero-read taxa

```{r}
summary(taxa_sums(physeq))
```

Save phyloseq object

```{r}
saveRDS(physeq, phyloseq_object)
```

How many samples, ASVs, and reads are there in the full sequenced dataset, respectively?

```{r}
nsamples(physeq)
ntaxa(physeq)
sum(sample_sums(physeq))
```

How many samples, ASVs, and reads are there in the wild sea otter dataset, respectively?  
- Wild (CDFW) sea otters only  
- No capture duplicates  
- No sea otters w/o official BRD number 
- Oral, rectal, seawater samples (no fecal) 

```{r}
totalCDFW2 <- subset_samples(physeq, facility=="CDFW" & species=="sea_otter" & duplicate_capture == "no")
totalCDFW2 <- subset_samples(totalCDFW2, sample_type == "oral" | sample_type == "rectal" | sample_type == "water")
totalCDFW2 = filter_taxa(totalCDFW2, function(x) sum(x) > 0, TRUE)
nsamples(totalCDFW2)
ntaxa(totalCDFW2)
sum(sample_sums(totalCDFW2))
```

How many distinct wild sea otters are there in the study?
- Wild (CDFW) sea otters only  
- No capture duplicates  
- No sea otters w/o official BRD number  

```{r}
num_unique_so <- sample_data(totalCDFW2)[, "brd_number"]
lengths(unique(num_unique_so))
```

During which years were samples collected?

```{r}
table(sample_data(totalCDFW2)$year_captured)
```

From which locations where sea otters sampled/captured?  
Note: this is the number of samples, not sea otters, per location.

```{r}
table(sample_data(totalCDFW2)$capture_location)
```

How many samples, ASVs, and reads are there in gingival samples from the wild sea otter dataset, respectively?  
- CDFW sea otters only  
- No capture duplicates  
- No sea otters w/o official BRD number 
 
```{r}
oralCDFW <- subset_samples(physeq, facility=="CDFW" & species=="sea_otter" & duplicate_capture == "no" & sample_type == "oral")
oralCDFW = filter_taxa(oralCDFW, function(x) sum(x) > 0, TRUE)
nsamples(oralCDFW)
ntaxa(oralCDFW)
sum(sample_sums(oralCDFW))
```

How many samples, ASVs, and reads are there in rectal samples from the wild sea otter dataset, respectively?  
- CDFW sea otters only  
- No capture duplicates  
- No sea otters w/o official BRD number
  
```{r}
rectalCDFW <- subset_samples(physeq, facility=="CDFW" & species=="sea_otter" & duplicate_capture == "no" & sample_type == "rectal")
rectalCDFW = filter_taxa(rectalCDFW, function(x) sum(x) > 0, TRUE)
nsamples(rectalCDFW)
ntaxa(rectalCDFW)
sum(sample_sums(rectalCDFW))
```

How many samples, ASVs, and reads are there in seawater samples from the wild sea otter dataset, respectively (using only samples associated with a sea otter in the study)?
- CDFW sea otters only  
- No capture duplicates  
- No sea otters w/o official BRD number  

```{r}
waterCDFW <- subset_samples(physeq, facility=="CDFW" & species=="sea_otter" & duplicate_capture == "no" & sample_type == "water")
waterCDFW = filter_taxa(waterCDFW, function(x) sum(x) > 0, TRUE)
nsamples(waterCDFW)
ntaxa(waterCDFW)
sum(sample_sums(waterCDFW))
```

Calculate relative abundance of genera in gingival and rectal samples -> req'd for Figures 2A, B

```{r}
physeq_relAb = transform_sample_counts(totalCDFW2, function(x) x/sum(x)*100)
physeq_phylum = tax_glom(physeq_relAb, taxrank=rank_names(physeq_relAb)[6])

physeq_genus_oral <- subset_samples(physeq_phylum, sample_type=="oral")
physeq_genus_oral = filter_taxa(physeq_genus_oral, function(x) sum(x) > 0, TRUE)

physeq_genus_rectal <- subset_samples(physeq_phylum, sample_type=="rectal")
physeq_genus_rectal = filter_taxa(physeq_genus_rectal, function(x) sum(x) > 0, TRUE)

write.csv(otu_table(physeq_genus_oral), asv_oral, quote=FALSE)
write.csv(tax_table(physeq_genus_oral), taxa_oral, quote=FALSE)
write.csv(otu_table(physeq_genus_rectal), asv_rectal, quote=FALSE)
write.csv(tax_table(physeq_genus_rectal), taxa_rectal, quote=FALSE)
```
